modules/ocf/manifests/configbuilder.pp

```diff
@@ -0,0 +1,22 @@
+define ocf::configbuilder(
+  $layout = {},
+  $owner = 'root',
+  $group = 'root',
+  $mode = '0400',
+  $show_diff = false,
+  $force = true,
+) {
+
+  $builder = $layout.tree_each.map |$h| { [ $h[0], ocf::lookup_wrap($h[1]) ] }
```

Comment:
Would it be possible to just have the layout that is passed in already have this format? Enforcing it has a type of `Hash[String, Hash[String, String]]` I believe should do the trick.